src/routes/v1/addresses.ts

```diff
@@ -2,10 +2,19 @@ import * as db from '../../utils/db';
 import { Request, Response } from 'express';
 
 export default async (req: Request, res: Response) => {
-    const result = {};
-    const addresses: any = await db.all(
+    const query = [
         'SELECT entries.*, addresses.address FROM addresses LEFT JOIN entries ON entries.id = addresses.entry'
-    );
+    ];
+    if (req.query.ticker) {
+        const ticker = req.query.ticker;
+        // Validate and sanitize input
+        if (ticker.match(/^[a-z]{3,5}$/i)) {
+            query.push(`WHERE addresses.coin = '${ticker.toUpperCase()}'`);
```

Comment:
Can you use a query variable here?

Reply:
Added prepared statement in commit https://github.com/CryptoScamDB/api.cryptoscamdb.org/pull/101/commits/eb6c5e0fb01ac8a6cd3c8575a2dfafc2d8a4695f